### PR TITLE
[SYCLomatic] Add 38 cuRAND examples for query-api-mapping.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by zhiweij1 — APPROVED** [review]

File: clang/test/dpct/query_api_mapping/cuRAND/test.cu [hunk 1]
```diff
@@ -0,0 +1,651 @@
...
+// CURAND-NEXT:   curand(ps1 /*curandStateMtgp32_t **/);
+// CURAND-NEXT:   curand(ps2 /*curandStateScrambledSobol64_t **/);
+// CURAND-NEXT:   curand(ps3 /*curandStateSobol64_t **/);
+// CURAND-NEXT:   curand(ps4 /*curandStateScrambledSobol32_t **/);
+// CURAND-NEXT:   curand(ps5 /*curandStateSobol32_t **/);
+// CURAND-NEXT:   curand(ps6 /*curandStateMRG32k3a_t **/);
+// CURAND-NEXT:   curand(ps7 /*curandStatePhilox4_32_10_t **/);
+// CURAND-NEXT:   curand(ps8 /*curandStateXORWOW_t **/);
+// CURAND-NEXT: Is migrated to:
+// CURAND-NEXT:   curandStateMtgp32_t *ps1;
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/cuRAND/test.cu]: L173~L177 is cuda code,  it is not correct.

File: clang/test/dpct/query_api_mapping/cuRAND/test.cu [hunk 2]
```diff
@@ -0,0 +1,651 @@
...
+// CURAND_LOG_NORMAL-NEXT:   curand_log_normal(ps5 /*curandStateMtgp32_t **/, f1 /*float*/, f2 /*float*/);
+// CURAND_LOG_NORMAL-NEXT:   curand_log_normal(ps6 /*curandStateMRG32k3a_t **/, f1 /*float*/,
+// CURAND_LOG_NORMAL-NEXT:                     f2 /*float*/);
+// CURAND_LOG_NORMAL-NEXT:   curand_log_normal(ps7 /*curandStatePhilox4_32_10_t **/, f1 /*float*/,
+// CURAND_LOG_NORMAL-NEXT:                     f2 /*float*/);
+// CURAND_LOG_NORMAL-NEXT:   curand_log_normal(ps8 /*curandStateXORWOW_t **/, f1 /*float*/, f2 /*float*/);
+// CURAND_LOG_NORMAL-NEXT: Is migrated to:
+// CURAND_LOG_NORMAL-NEXT:   curandStateScrambledSobol64_t *ps1;
+// CURAND_LOG_NORMAL-NEXT:   curandStateSobol64_t *ps2;
+// CURAND_LOG_NORMAL-NEXT:   curandStateScrambledSobol32_t *ps3;
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/cuRAND/test.cu]: Same as above.

**Review by zhiweij1 — CHANGES_REQUESTED** [review]
Please remove the unsupported APIs (which are using types: `curandStateMtgp32_t`, `curandStateScrambledSobol64_t`, `curandStateSobol64_t`, `curandStateScrambledSobol32_t`, `curandStateSobol32_t`

File: clang/test/dpct/query_api_mapping/cuRAND/test.cu (hunk 1 above)
Comment (tangjj11) [reply clang/test/dpct/query_api_mapping/cuRAND/test.cu]: deleted

File: clang/test/dpct/query_api_mapping/cuRAND/test.cu (hunk 2 above)
Comment (tangjj11) [reply clang/test/dpct/query_api_mapping/cuRAND/test.cu]: deleted

**Review by zhiweij1 — APPROVED** [review]

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**